internal/exec/stages/disks/filesystems.go

```diff
@@ -108,7 +108,7 @@ func (s stage) createFilesystem(fs types.Mount) error {
 		// If the filesystem isn't forcefully being created, then we need
 		// to check if it is of the correct type or that no filesystem exists.
 
-		if (info.format == fs.Format || (info.label == "OEM" && *fs.Label == "OEM")) &&
+		if (info.format == fs.Format || (info.label == "OEM" && fs.Label != nil && *fs.Label == "OEM")) &&
```

Comment:
I think this still has a problem if you don't specify fs.Label, because then the right side of the '||' is automatically false, so we continue to `return ErrBadFilesystem`. I think what you wanted to express was `if (info.format == fs.Format || info.label == "OEM") because the next line `(fs.Label == nil || info.label == *fs.Label)` takes care of ensuring that they labels match.

Reply:
I wanted the logic to reuse the OEM partition to be limited to cases where the user specifies a valid new OEM partition. With your suggestion we would always reuse the OEM partition and not allow the user to replace it with something else.

Reply:
But we're in the `if !fs.WipeFilesystem` case.

Reply:
Ok, sounds good :+1:

Reply:
Updated the commit